src/Processors/Transforms/SquashingChunksTransform.cpp

```diff
@@ -64,8 +69,22 @@ void SimpleSquashingChunksTransform::transform(Chunk & chunk)
     }
     else
     {
+        if (chunk.hasRows())
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Chunk expected to be empty, otherwise it will be lost");
+
         auto block = squashing.add({});
         chunk.setColumns(block.getColumns(), block.rows());
+
+        /// ISimpleTransform keeps output chunk (result of transform() execution) for some time and push it in the output port within subsequent prepare() call.
+        /// Because of our custom prepare() implementation we have to take care of both places where data could be buffered: `output_data` and `squashing`.
+        if (output_data.chunk.hasRows())
```

Comment:
I find this rather hacky. IMO, the proper fix would be to modify `prepair()` i.e. check there as well if we can push to the output port, i.e. we can simply write left data to `chunk` as usual.